posts/airflow-ray.md

```diff
@@ -0,0 +1,285 @@
+
+## Introducing Ray
+
+[Ray](https://ray.io/) is a cluster computing framework built with Python as a first-class citizen. Unlike systems like Spark, which require a complex cluster set-up and Java dependencies, Ray can run the same code on any machine from a user’s laptop to a highly powered AWS virtual machine with very minimal configuration. 
+
+Once running, users can allocate Ray resources on a per-function basis (e.g. give this function 2 CPUs and 1 GPU). If used correctly, Ray’s distributed computation combined with Airflow’s robust scheduling and orchestration make a perfect platform for rapid, reliable ML development and deployment.
+
+Ray is highly performant, and under the hood is written in C++ to quickly & automatically move Python objects around the cluster using gRPC as new functions are called that require data from previous calls. All of this is abstracted away — as a user, you simply write Python code. 
+
+
+## Airflow + Ray: A match made in data science heaven
+
+Airflow and Ray need no special packages or setup to work together, but we found after working with many users, the same patterns came up again and again. We wanted to encapsulate best practices in a way that can standardize Airflow + Ray usage in a tested, scalable way.
+
+Together with the [Ray](https://ray.io/) and [Anyscale](https://www.anyscale.com/) teams, we are excited to introduce you  to the [Ray Provider for Apache Airflow](https://registry.astronomer-stage.io/providers/ray). 
+
+In this provider, we encapsulate all the Ray-specific setup/initialization code into decorators instead of an Operator by extending the [TaskFlow API](https://airflow.apache.org/docs/apache-airflow/stable/tutorial_taskflow_api.html#example-taskflow-api-etl-pipeline). This decorator allows Airflow users to keep all of their Ray code in Python functions and define task dependencies by moving data through python functions.
+
+As a result, Airflow + Ray users will know the code they are launching and have complete flexibility to modify and templatize their DAGs while still taking advantage of Ray’s distributed computation capabilities.
+
+To show the power of this integration, let’s start our data science pipeline where all data science pipelines start: a Jupyter notebook.
+
+## Training an XGBoost model with Ray in a notebook
+
+In this example, we’ve created a basic Jupyter notebook model that pulls the HIGGS dataset, splits training and testing data, and creates/validates a model using the XGBoost on Ray, which scales XGBoost training onto a cluster using Ray. Everything here can run locally on a user’s laptop or via a remote Ray cluster with minimal work. This stack handles the experimentation aspect of model building very well, and we can now dive into how to turn his experiment into a production-ready Airflow DAG.
+
+
+![Ray Notebook](../assets/ray_notebook.png)
```

Comment:
Broken image link here. I couldn't find this file in the assets directory. PLEASE make sure this works before publishing. You should be able to see the image appear when looking at this doc in the "Rich Diff" view.